### PR TITLE
Add pre-trained weights for ViTs

### PR DESCRIPTION
**Pull request by DavidLandup0**
# What does this PR do?

Converted to draft until #1014 is merged.

Since the last PR (#1014) is already getting large, and the description is verbose - separating the weights into a new PR.
Cloned from the ViT branch (once that's merged, this PR will be rebased and most of the changes in the files will disappear as they're already included in the previous PR).

Here are the official JAX weights, ported for KCV ViTs, as H5 files:

- Tiny16: https://drive.google.com/file/d/1fEo0l_zwibl3eJ6-xb9MTvHgIo8XySxY
- S16: https://drive.google.com/file/d/1R1tgxJScy0LK3HLGNYWwLG_UQp10ope2
- B16: https://drive.google.com/file/d/1M_zALLuEuoZ1fZNpWPqlD2Ahc9UPB9G4
- L16: https://drive.google.com/file/d/1QcPYoDZMbIaT-EdW6CjjqiwDaMbML9Zu
- S32: https://drive.google.com/file/d/1p7DCy3eukJFuzqbgdXWILNhJUx2e8Kzp
- B32: https://drive.google.com/file/d/1M_zALLuEuoZ1fZNpWPqlD2Ahc9UPB9G4

Updated from GCS to Google Drive to avoid bucket cost accumulation.

Inference on the ImageNetV2 validation set (ms/sep, top-1 and top-5 ACC):

Tiny16: 87ms/step - loss: 1.6895 - accuracy: 0.6067 - sparse_top_k_categorical_accuracy: 0.8285
S16:      178ms/step - loss: 1.2600 - accuracy: 0.6884 - sparse_top_k_categorical_accuracy: 0.8916
B16:      464ms/step - loss: 1.0631 - accuracy: 0.7293 - sparse_top_k_categorical_accuracy: 0.9199
L16:      1s/step - loss: 0.9622 - accuracy: 0.7545 - sparse_top_k_categorical_accuracy: 0.9321
S32:      54ms/step - loss: 1.6650 - accuracy: 0.6090 - sparse_top_k_categorical_accuracy: 0.8335
B32:      122ms/step - loss: 1.3684 - accuracy: 0.6650 - sparse_top_k_categorical_accuracy: 0.8781

The PR also updates `vit.py` to `parse_weights()` using their aliases - though, there are no hashes yet. Can you help with uploading them @ianstenbit?

### CODE REVIEW

**Comment by ianstenbit** [comment]
Thanks @DavidLandup0!

So far, all of our KerasCV weights are trained "in-house" (by our training scripts), and if possible we'd like to keep it that way (although I understand this may turn out to be very challenging for ViTs).

I'm going to sync with the team on this and get back to you -- it may be that offering the official weights for these models in particular is a good approach.

**Comment by DavidLandup0** [comment]
I agree that we want to prefer reproducible weights with scripts hosted here. With a bit of back and forth in the comments of the last PR, we figured that porting them would probably be okay for these, at least for the larger ones 600M+ params).

I'm still working on the training script for ViTs so we have both a script for training *and* the free converted weights (eco friendly, yay!), but they're tricky to figure out as the divergence happens a couple of hours in, so every test run takes hours of good-looking curves followed by plateauing :(

As soon as I can compile a good script for training these, I'll submit it here as well :)

**Comment by DavidLandup0** [comment]
P.S. These weights are pre-trained on ImageNet21k and fine-tuned on 1k, while our training script operates on 1k only currently.

**Comment by DavidLandup0** [comment]
Agreed with @bhack. ViTs are tricky to train without a lot of data, so in their plain form, they're the most useful when using pre-trained official weights. 

There are few runs online to compare to (expensive + subpar accuracy without ImageNet21k pretraining), but there are pre-trained runs that we can compare to, on much more manageable datasets like CIFAR10 and CIFAR100 (happy to run a bunch of these on my local machine). Since users will be using them with pre-trained weights (otherwise, they can achieve better results with fewer flops using a CNN like EfficientNet), I think we should first validate the fine-tuning of this network with official weights, and then work on reproducing them using an in-house script later when that's more feasible.

Thoughts?
